modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase

```diff
@@ -1019,6 +1019,124 @@ definition {
+	test CanCreateValidationUsingOldValueFunctionWithPicklistField {
+		task ("Given: A picklist with two items is created (via API)") {
+			Picklist.addPicklistViaAPI(picklistName = "Status");
+
+			Picklist.addPicklistItemViaAPI(
+				itemKey = "open",
+				itemName = "Open",
+				picklistName = "Status");
+
+			Picklist.addPicklistItemViaAPI(
+				itemKey = "inProgress",
+				itemName = "In Progress",
+				picklistName = "Status");
+		}
+
+		task ("And given: An object with a field is created") {
+			ObjectAdmin.addObjectViaAPI(
+				labelName = "Custom Object 175196",
+				objectName = "CustomObject175196",
+				pluralLabelName = "Custom Objects 175196");
+
+			ObjectAdmin.addObjectFieldViaAPI(
+				fieldBusinessType = "Picklist",
+				fieldLabelName = "Custom Picklist Field",
+				fieldName = "customPicklistField",
+				fieldType = "String",
+				isRequired = "false",
+				objectName = "CustomObject175196",
+				picklistName = "Status");
+		}
+
+		task ("And given: A custom validation with expression builder is created and the object is published") {
+			ObjectAdmin.openObjectAdmin();
+
+			ObjectPortlet.selectCustomObject(label = "Custom Object 175196");
+
+			ObjectAdmin.goToValidationsTab();
+
+			ObjectCustomValidation.addObjectValidation(
+				validationName = "Custom Validation",
+				validationType = "Expression Builder");
+
+			ObjectCustomValidation.gotoCustomValidation(validationName = "Custom Validation");
+
+			ObjectCustomValidation.gotoTab(tabName = "Conditions");
+
+			ObjectCustomValidation.insertScriptOnValidation(scriptValidation = '''oldValue("customPicklistField") == "inProgress"''');
+
+			ObjectCustomValidation.addErrorMessage(errorMessage = "Invalid Value.");
+
+			PortletEntry.save();
+
+			ObjectAdmin.publishObjectViaAPI(objectName = "CustomObject175196");
+		}
+
+		task ("When: An entry is created") {
+			ObjectAdmin.goToCustomObject(objectName = "CustomObject175196");
+
+			ObjectAdmin.addObjectPicklistFieldEntryViaUI(picklistOption = "Open");
+		}
+
+		task ("And when: the validation toggle is active") {
+			ObjectAdmin.openObjectAdmin();
+
+			ObjectPortlet.selectCustomObject(label = "Custom Object 175196");
+
+			ObjectAdmin.goToValidationsTab();
+
+			ObjectCustomValidation.gotoCustomValidation(validationName = "Custom Validation");
+
+			ObjectCustomValidation.activeValidation();
+
+			PortletEntry.save();
+		}
+
+		task ("Then: The object entry is edited and the error message is displayed") {
+			ObjectAdmin.goToCustomObject(objectName = "CustomObject175196");
+
+			ObjectPortlet.viewEntryDetails(entry = "Open");
+
+			ObjectAdmin.changePicklistFieldEntry(
+				entrySelected = "Open",
+				invalidEntry = "true",
+				newPicklistOption = "In Progress");
+
+			Alert.viewErrorMessage(errorMessage = "Invalid Value.");
+		}
+
+		task ("And then: The expression is edited with a new condition and the entry is updated") {
```

Comment:
Hey you did a really good job with this section, but I have a slight issue with the test steps in general. For automation tests we try not to have multiple asserts like this that require a lot more additional setup after the first assert. This looks like it could almost be 2 separate tests. My proposal to change this test would be to 

1. In your first block; create 3 picklist items
2. Keep everything the same until your "Then"
3. After the Invalid Value assertion; edit the entry to be the name of the 3rd picklist item you created
4. Assert that entry is updated

Reply:
Possibly separate into two tests instead